packages/dev/core/src/Shaders/color.vertex.fx

```diff
@@ -47,10 +47,14 @@ void main(void) {
 
 #include<clipPlaneVertex>
 
-#ifdef VERTEXCOLOR
-	vColor = color;
-#elif INSTANCESCOLOR
-	vColor = instanceColor;
+#if defined(VERTEXCOLOR) || defined(INSTANCESCOLOR)
+    vColor = vec4(1.0);
+    #ifdef VERTEXCOLOR
+        vColor *= color;
+    #endif
+    #ifdef INSTANCESCOLOR
+        vColor *= instanceColor;
+    #endif

```

Comment:
What about putting this block in an external file that you can #include?

It is reused a lot of time, and if some changes should occur they will have to be done for all the blocks.